tokio-util/src/io/sync_bridge.rs

```diff
@@ -0,0 +1,82 @@
+use pin_project_lite::pin_project;
+use std::io::{Read, Write};
+use std::pin::Pin;
+use tokio::io::{AsyncRead, AsyncReadExt, AsyncWrite, AsyncWriteExt};
+
+pin_project! {
+    /// Use a [`tokio::io::AsyncRead`] synchronously as a [`std::io::Read`] or
+    /// a [`tokio::io::AsyncWrite`] as a [`std::io::Write`].
+    #[derive(Debug)]
+    pub struct SyncIoBridge<T> {
+        src: Pin<Box<T>>,
```

Comment:
I would prefer to not box it. If the user has a non-Unpin type, we can provide a `new_pin` method that takes a `T` and returns a `SyncIoBridge<Pin<Box<T>>>`.
```suggestion
        src: T,
```

Reply:
Well, I'll be honest; I'm still mostly at the "know enough that I can get stuff to compile and usually work" with async Rust.  I definitely throw in a `Box` here or there when I hit issues.  I've read https://fasterthanli.me/articles/pin-and-suffering twice and feel I'll need a 3rd time soon.

The previous use of `pin_project` here is a legacy of my attempts to not box and...
[time passes] ⏲

Ah duh, OK it all works by adding an `+ Unpin` bound.  Fixed.  Thanks for the review!  Saving the world, one fewer heap allocation at a time.

Reply:
While I have this comment thread open, did you have any opinions on the naming https://github.com/tokio-rs/tokio/pull/4146#discussion_r720863960 ?  I am kind of leaning towards `StdIoBridge` currently.